src/Shared/FileUtilities.cs

```diff
@@ -59,19 +67,16 @@ static internal partial class FileUtilities
         private static void GetTestExecutionInfo()
         {
             // Get the executable we are running
-            var program = Path.GetFileNameWithoutExtension(Environment.GetCommandLineArgs()[0]);
+            var processNameCommandLine = Path.GetFileNameWithoutExtension(Environment.GetCommandLineArgs()[0]);
+            var processNameCurrentProcess = Process.GetCurrentProcess().ProcessName;
```

Comment:
Honestly no idea. The previous comments were completely unhelpful for that. I assume the correct mechanism varies depending on how the process was created, but given that this code was 100% wrong before I can't really back that up. The values are generally the same from the tests I did.